tests/reg_tests/test_DVGeometry.py

```diff
@@ -1460,6 +1460,72 @@ def test_shape_functions(self, train=False, refDeriv=False):
 
             handler.root_add_val("new_pts", new_pts, rtol=1e-10, atol=1e-10)
 
+    def train_active_children(self, train=True):
+        self.test_active_children(train=train)
+
+    def test_active_children(self, train=False):
+        """
+        Test active children option for adding pointsets
+        """
+
+        refFile = os.path.join(self.base_path, "ref/test_active_children.ref")
+        with BaseRegTest(refFile, train=train) as handler:
+            handler.root_print("Test shape function DVs")
+
+            DVGeo, DVGeoChild1 = commonUtils.setupDVGeo(self.base_path, childName="child1")
+            _, DVGeoChild2 = commonUtils.setupDVGeo(self.base_path, childName="child2")
+
+            # add design variables
+            DVGeoChild1.addGlobalDV(dvName="span1", value=0.5, func=commonUtils.spanX, lower=0.1, upper=10, scale=1)
+            DVGeoChild2.addGlobalDV(dvName="span2", value=0.5, func=commonUtils.spanX, lower=0.1, upper=10, scale=1)
+            DVGeo.addChild(DVGeoChild1)
+            DVGeo.addChild(DVGeoChild2)
+
+            points = np.zeros([2, 3])
+            points[0, :] = [0.25, 0, 0]
+            points[1, :] = [-0.25, 0, 0]
+
+            # first, all children active
+            ptName = "testPointsAll"
+            DVGeo.addPointSet(points, ptName)
+
+            # only the first child
+            ptName = "testPoints1"
+            DVGeo.addPointSet(points, ptName, activeChildren=["child1"])
+
+            # only second
+            ptName = "testPoints2"
+            DVGeo.addPointSet(points, ptName, activeChildren=["child2"])
+
+            # no children
+            ptName = "testPointsNone"
+            DVGeo.addPointSet(points, ptName, activeChildren=[])
+
+            nPt = points.size
+            ptNames = ["testPointsAll", "testPoints1", "testPoints2", "testPointsNone"]
+            for ptName in ptNames:
+                # test derivatives
+                dIdPt = np.zeros([nPt, 2, 3])
+                dIdPt[0, 0, 0] = 1.0
+                dIdPt[1, 0, 1] = 1.0
+                dIdPt[2, 0, 2] = 1.0
+                dIdPt[3, 1, 0] = 1.0
+                dIdPt[4, 1, 1] = 1.0
+                dIdPt[5, 1, 2] = 1.0
+                dIdx = DVGeo.totalSensitivity(dIdPt, ptName)
+
+                handler.root_add_dict(f"dIdx_{ptName}", dIdx, rtol=1e-10, atol=1e-10)
```

Comment:
Looks like the test is failing here with `KeyError: 'child1_span1'` but once that's fixed I'm good with this PR